build.zig

```diff
@@ -1229,30 +1229,6 @@ const FailStep = struct {
     }
 };
 
-// Patch the target to use the right CPU. This is a somewhat hacky way to do this, but the core idea
-// here is to keep this file as the source of truth for what we need from the CPU.
-fn set_cpu_features(
```

Comment:
We no longer patch the target and instead just select the correct target in the first place.